src/aiida_quantumespresso/workflows/xspectra/core.py

```diff
@@ -0,0 +1,910 @@
+# -*- coding: utf-8 -*-
+"""Workchain to compute the X-ray absorption spectrum for a given structure.
+
+Uses QuantumESPRESSO pw.x and xspectra.x, requires ``AiiDA-Shell`` to run ``upf2plotcore.sh``.
+"""
+import pathlib
+from typing import Optional, Union
+
+from aiida import orm
+from aiida.common import AttributeDict
+from aiida.engine import ToContext, WorkChain, calcfunction, if_
+from aiida.orm.nodes.data.base import to_aiida_type
+from aiida.plugins import CalculationFactory, DataFactory, WorkflowFactory
+import yaml
+
+from aiida_quantumespresso.utils.mapping import prepare_process_inputs
+from aiida_quantumespresso.workflows.protocols.utils import ProtocolMixin, recursive_merge
+
+PwCalculation = CalculationFactory('quantumespresso.pw')
+PwBaseWorkChain = WorkflowFactory('quantumespresso.pw.base')
+XspectraBaseWorkChain = WorkflowFactory('quantumespresso.xspectra.base')
+XyData = DataFactory('array.xy')
+
+
+@calcfunction
+def get_all_spectra(**kwargs):
+    """Compile all calculated spectra into a single XyData node for easier plotting.
```

Comment:
Can you please describe in the docstring what nodes will be accepted/expected in `**kwargs`. For example, should be `XyData` nodes that are the `output_spectra` of an `XspectraCalculation`. Not sure if these labels are correct, but just an example.

Reply:
Something like this, potentially?

    """Compile all calculated spectra into a single XyData node for easier plotting.

    The keyword arguments must be an arbitrary number of XyData nodes from 
    the `output_spectra` of `XspectraCalculation`s, all other `kwargs` will be discarded at 
    runtime.
    
    Returns a single XyData node where each set of y values is labelled
    according to the polarisation vector used for the `XspectraCalculation`.
    """